code/framework/encryption/src/main/java/io/cattle/platform/framework/encryption/handler/impl/TransformerHandlerImpl.java

```diff
@@ -0,0 +1,44 @@
+package io.cattle.platform.framework.encryption.handler.impl;
+
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.framework.encryption.handler.TransformerHandler;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class TransformerHandlerImpl extends TransformerHandler {
+
+    private static final DynamicStringProperty ENCRYPTER_NAME = ArchaiusUtil.getString("api.encryption.encrypter");
+    private static final DynamicStringProperty HASHER_NAME = ArchaiusUtil.getString("api.encryption.hasher");
+
+    @Override
+    public String transform(String value, String method) {
+        String name;
+        if (method.equalsIgnoreCase("encrypt")){
+            name = ENCRYPTER_NAME.getValue();
+        } else if (method.equalsIgnoreCase("hash")) {
+            name = HASHER_NAME.getValue();
+        } else {
+            throw new IllegalArgumentException(method);
+        }
+        if (!getTransformers().containsKey(name)){
+            throw new IllegalArgumentException(name);
+        }
+        return name + TransformerHandler.ENCRYPTER_NAME_DELM + getTransformers().get(name).transform(value);
+    }
+
+    @Override
+    protected String untransformInternal(String transformerName, String value) {
+        if (!getTransformers().containsKey(transformerName)){
```

Comment:
Nit: we can extract lines 31:33 to its own method. As this logic is being re-used on lines 39:41